Docs: Update 9.3.4 known issue with impact details (#15056)
elastic/elastic-agent: docs/release-notes/known-issues.md

For more information check [Issue #264983](https://github.com/elastic/kibana/issues/264983).
:::

:::{dropdown} Events from Beats-based inputs and integrations in Elastic Agent 9.3.4 incorrectly convert timestamps to an empty {} JSON object.
:name: events-beats-9-3-4-timestamp-empty-object
**Applies to: {{agent}} 9.3.4**

> A performance optimization in {{agent}} 9.3.4 causes timestamp fields produced by Beats-based inputs to be incorrectly serialized as an empty `{}` JSON object. This affects time fields in the event body (for example, `event.created`) but does not affect the primary `@timestamp` field.

Author: ```suggestion
Due to the broad impact of this issue, {{agent}} 9.3.4 has been removed from the downloads page. We recommend that you skip {{agent}} 9.3.4 and upgrade directly to 9.3.5 or later.

A performance optimization in {{agent}} 9.3.4 causes timestamp fields produced by Beats-based inputs to be incorrectly serialized as an empty `{}` JSON object. This affects time fields in the event body (for example, `event.created`) but does not affect the primary `@timestamp` field.
```

For more information check [Issue #264983](https://github.com/elastic/kibana/issues/264983).
:::

:::{dropdown} Events from Beats-based inputs and integrations in Elastic Agent 9.3.4 incorrectly convert timestamps to an empty {} JSON object.
:name: events-beats-9-3-4-timestamp-empty-object
**Applies to: {{agent}} 9.3.4**

> Due to the broad impact of this issue, {{agent}} 9.3.4 has been removed from the downloads page. We recommend that you skip {{agent}} 9.3.4 and upgrade directly to 9.3.5 or later.

Author: ```suggestion
:::{warning}
Due to the broad impact of this issue, {{agent}} 9.3.4 has been removed from the downloads page. We recommend that you skip {{agent}} 9.3.4 and upgrade directly to 9.3.5 or later.
:::
```

For more information check [Issue #264983](https://github.com/elastic/kibana/issues/264983).
:::

> :::{dropdown} Events from Beats-based inputs and integrations in Elastic Agent 9.3.4 incorrectly convert timestamps to an empty {} JSON object.

Author: ```suggestion
::::{dropdown} Events from Beats-based inputs and integrations in Elastic Agent 9.3.4 incorrectly convert timestamps to an empty {} JSON object.
```

**Impact**

Because most {{agent}} integrations and inputs rely on Beats-based collectors, the impact is wide-ranging. Examples of affected functionality include:

* **Synthetics monitoring** — Synthetics monitors rely on Heartbeat and may stop reporting expected data, resulting in a loss of monitoring visibility.
* **SentinelOne integration** — Response actions stop functioning when `event.created` is missing. For details, refer to [Issue #266355](https://github.com/elastic/kibana/issues/266355).
* **Other data-collection integrations** — Any integration that sends events through Beats-based inputs may emit affected documents.

**Symptoms**

Documents from Beats-based inputs contain one or more time fields serialized as an empty object, for example:

```json
"event": {
  "dataset": "sentinel_one.agent",
  "created": {}
}
```

**Resolution**

The performance optimization has been removed. The fix is included in {{agent}} 9.3.5. This issue does not affect {{agent}} 9.4.0.

> :::

Author: ```suggestion
::::
```

For more information check [Issue #264983](https://github.com/elastic/kibana/issues/264983).
:::

::::{dropdown} Events from Beats-based inputs and integrations in Elastic Agent 9.3.4 incorrectly convert timestamps to an empty {} JSON object.
:name: events-beats-9-3-4-timestamp-empty-object
**Applies to: {{agent}} 9.3.4**

:::{warning}

> Due to the broad impact of this issue, {{agent}} 9.3.4 has been removed from the downloads page. We recommend that you skip {{agent}} 9.3.4 and upgrade directly to 9.3.5 or later.

Reviewer: I think it's "Downloads" page, but please double-check the capitalization. 

Also would change to "We recommend upgrading directly to 9.3.5 or later" because if it's removed from the Downloads page, there's nothing to "skip" (because it doesn't exist) if that makes sense.

Author: The page is not actually called “Downloads"… 
This is where you’d download the latest version: [Download Elastic Agent](https://www.elastic.co/downloads/elastic-agent)
This is where you’d find the links to previous releases (with a “Download" button): [Past releases](https://www.elastic.co/downloads/past-releases#elastic-agent)
And this is the page where normally the links to the older EA version would be: [Elastic Agent 9.3.4](https://www.elastic.co/downloads/past-releases/elastic-agent-9-3-4)

So I think we can keep it “downloads";)

> Due to the broad impact of this issue, {{agent}} 9.3.4 has been removed from the downloads page. We recommend that you skip {{agent}} 9.3.4 and upgrade directly to 9.3.5 or later.

Author: ```suggestion
Due to the broad impact of this issue, {{agent}} 9.3.4 has been removed from the downloads page. We recommend upgrading directly to {{agent}} 9.3.5 or later.
```

:::

A performance optimization in {{agent}} 9.3.4 causes timestamp fields produced by Beats-based inputs to be incorrectly serialized as an empty `{}` JSON object. This affects time fields in the event body (for example, `event.created`) but does not affect the primary `@timestamp` field.

**Impact**

Because most {{agent}} integrations and inputs rely on Beats-based collectors, the impact is wide-ranging. Examples of affected functionality include:

> * **Synthetics monitoring** — Synthetics monitors rely on Heartbeat and may stop reporting expected data, resulting in a loss of monitoring visibility.

Reviewer 2: ```suggestion
* **Synthetics monitoring** — Synthetics monitors rely on Heartbeat and might stop reporting expected data, resulting in a loss of monitoring visibility.
```
